service/history/ndc/branch_manager.go

```diff
@@ -249,7 +254,11 @@ func (r *branchManagerImpl) createNewBranch(
 	if err := newVersionHistory.SetBranchToken(resp.NewBranchToken); err != nil {
 		return 0, err
 	}
-	branchChanged, newIndex, err := r.mutableState.GetVersionHistories().AddVersionHistory(
+	versionHistory := r.mutableState.GetVersionHistories()
+	if versionHistory == nil {
+		return 0, &shared.BadRequestError{Message: "Current reapply does not support 2dc workflow."}
```

Comment:
Is returning a custom error instead of `execution.ErrMissingVersionHistories` intentional?

Reply:
updated